tests/test_processor.py

```diff
@@ -328,3 +400,120 @@ def test_mixed_deps_two_apps(mock_repo_file, processor, optional_deps_method, ex
+        select_all=True, components=["component1"], apps=["app1"]
+    )
+
+    assert (
+        _should_remove(remove_resources, no_remove_resources, "component2", "app2")
+        is False
+    )
+    assert (
+        _should_remove(remove_resources, no_remove_resources, "component1", "app2")
+        is False
+    )
+    assert (
+        _should_remove(remove_resources, no_remove_resources, "whatever", "app1")
+        is False
+    )
+
+
+def test_should_remove_remove_for_all_no_exceptions():
+    remove_resources = AppOrComponentSelector(
+        select_all=True, components=["component1"], apps=["app1"]
+    )
+    no_remove_resources = AppOrComponentSelector(
+        select_all=False, components=[], apps=[]
+    )
+
+    assert (
+        _should_remove(remove_resources, no_remove_resources, "component2", "app2")
+        is True
+    )
+    assert (
+        _should_remove(remove_resources, no_remove_resources, "component1", "app2")
+        is True
+    )
+    assert (
+        _should_remove(remove_resources, no_remove_resources, "whatever", "app1")
+        is True
+    )
+
+
+def test_should_remove_remove_option_select_all():
+    assert (
+        _should_remove(
+            AppOrComponentSelector(select_all=True, components=[], apps=[]),
+            AppOrComponentSelector(select_all=False, components=["component1"], apps=[]),
+            "app2",
+            "component1",
+        )
+        is False
+    )
+
+    assert (
+        _should_remove(
+            AppOrComponentSelector(select_all=True, components=[], apps=[]),
+            AppOrComponentSelector(select_all=False, components=[], apps=["app1"]),
+            "app1",
+            "component1",
+        )
+        is False
+    )
+
+    assert (
+        _should_remove(
+            AppOrComponentSelector(select_all=True, components=[], apps=[]),
+            AppOrComponentSelector(select_all=False, components=[], apps=[]),
+            "app1",
+            "component2",
+        )
+        is True
+    )
+
+
+def test_should_remove_no_remove_option_select_all():
+    assert (
+        _should_remove(
+            AppOrComponentSelector(select_all=False, components=["component1"], apps=[]),
+            AppOrComponentSelector(select_all=True, components=[], apps=[]),
+            "app1",
+            "component1",
+        )
+        is True
+    )
+
+    assert (
+        _should_remove(
+            AppOrComponentSelector(select_all=False, components=[], apps=["app1"]),
+            AppOrComponentSelector(select_all=True, components=[], apps=[]),
+            "app1",
+            "component1",
+        )
+        is True
+    )
+
+    assert (
+        _should_remove(
+            AppOrComponentSelector(select_all=False, components=["component1"], apps=["app1"]),
+            AppOrComponentSelector(select_all=True, components=[], apps=[]),
+            "app1",
+            "component1",
+        )
+        is True
+    )
+
```

Comment:
Can we add a test similar to the above (that uses the remove/no_remove options) but that asserts:
"app2" "component1" will return `True`
"app1" "component2" will return `True`
"app2" "component2" will return `False`